snap/info_snap_yaml.go

```diff
@@ -442,13 +442,23 @@ func validateAttr(v interface{}) (interface{}, error) {
 			if !ok {
 				return nil, fmt.Errorf("non-string key in attribute map: %v", k)
```

Comment:
we probably want to take  an argument to the function to distinguish "attribute"/"config default" or just "default"